src/models/CourseContract.ts

```diff
@@ -29,6 +29,10 @@ export class CourseContract extends BaseEntity {
   @Column(DataType.UUID)
   courseId: string;
 
+  @Default([])
+  @Column(DataType.JSON)
+  lessonProgress: string[];
```

Comment:
Why is the datatype `JSON` but default/type `string[]`

Reply:
cus the lessonProgress is an array of lessonIds. hence string[]

Reply:
```suggestion
  @Default([])
  @Column(DataType.ARRAY)
  lessonProgress: string[];
```

Wouldn't this be better?